src/hotspot/share/code/icBuffer.cpp

```diff
@@ -140,7 +140,12 @@ void ICStub::print() {
 
 void InlineCacheBuffer::initialize() {
   if (_buffer != NULL) return; // already initialized
-  _buffer = new StubQueue(new ICStubInterface, 10*K, InlineCacheBuffer_lock, "InlineCacheBuffer");
+  int code_size = align_up((int)(10*K), 2*BytesPerWord);
+  BufferBlob* blob = BufferBlob::create("InlineCacheBuffer", code_size);
+  if (blob == NULL) {
+    vm_exit_out_of_memory(code_size, OOM_MALLOC_ERROR, "CodeCache: no room for InlineCacheBuffer");
```

Comment:
Is VM exit here appropriate?
AFAICS, previously the failure to allocate InlineCacheBuffer was caught in assertion only.

Reply:
This is extracted from https://github.com/openjdk/jdk/pull/4337/files#diff-044ffb3850314d23822b79de889820f6432757df5431eac3748a9eb60b1b6899L69-L73.